sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAzureNamedKeyCredentialPolicy.java

```diff
@@ -131,7 +132,9 @@ private String getCanonicalizedResource(URL requestUrl) {
                 String queryParamValuesStr = String.join(",", queryParamValues);
```

Comment:
This should be pushed into the if statement below as if the statement is false this is never used, same with the sorting of the query parameter values.

Also, the value sort is in-place isn't it? Should this be done in an external container as to not mutate the parameters?

Reply:
I agree we can move the code you mentioned to be inside the if statement.

About the sorting, we create the map where both the query param names and values are contained from the request URL right before sorting and we don't really use these values anywhere else, so I think doing this in-place should be safe.